src/libraries/Native/Unix/System.Native/ios/net/if_media.h

```diff
@@ -0,0 +1,589 @@
+/*
+ * Copyright (c) 2000-2019 Apple Inc. All rights reserved.
+ *
+ * @APPLE_OSREFERENCE_LICENSE_HEADER_START@
+ *
+ * This file contains Original Code and/or Modifications of Original Code
+ * as defined in and that are subject to the Apple Public Source License
+ * Version 2.0 (the 'License'). You may not use this file except in
+ * compliance with the License. The rights granted to you under the License
+ * may not be used to create, or enable the creation or redistribution of,
+ * unlawful or unlicensed copies of an Apple operating system, or to
+ * circumvent, violate, or enable the circumvention or violation of, any
+ * terms of an Apple operating system software license agreement.
```

Comment:
Just want to make sure this license is compatible with our repo...

Reply:
I was using https://github.com/dotnet/runtime/blob/main/src/libraries/Native/Unix/System.Native/ios/net/route.h as precedent, but it does not hurt to double check.

/cc @richlander

Reply:
Just curious, why is it necessary to commit the Apple headers? Aren't they available in their SDK during build, similar to winnt.h etc?

@omajid do you consider this an acceptable license? I assume so since as @steveisok points out we already have such code.

Reply:
I wasn't sure, so I went looking.

https://www.gnu.org/philosophy/apsl.en.html says:

> The Apple Public Source License (APSL) version 2.0 qualifies as a free software license. 

It's in the OSI list too: https://opensource.org/licenses/APSL-2.0

Fedora considers this license as okay too: https://fedoraproject.org/wiki/Licensing:Main#Good_Licenses (search for "APSL 2.0" in the table). Red Hat legal has approved it for RHEL as well.

Reply:
> Just curious, why is it necessary to commit the Apple headers? Aren't they available in their SDK during build, similar to winnt.h etc?
> 
> @omajid do you consider this an acceptable license? I assume so since as @steveisok points out we already have such code.

It's unknown exactly why they are excluded from the iOS SDK. The likely explanation is that Apple can more easily move on from supporting them.